homeassistant/components/sensor/syncthru.py

```diff
@@ -115,6 +115,13 @@ def __init__(self, syncthru, name):
         self._name = name
         self._icon = 'mdi:printer'
         self._unit_of_measurement = None
+        self._id_suffix = ''
+
+    @property
+    def unique_id(self):
+        """Return unique ID for the sensor."""
+        serial = self.syncthru.serial_number()
+        return serial + self._id_suffix if serial else super().unique_id
```

Comment:
Just return `None` if unique_id is not available. Don't call super.

Reply:
Curious, why?

Why not: That would lose the benefit if the superclass implementation was sometime changed to return something else than `None`, changes would have to be made here as well. I think it's better to have it this way. Just usual OO style coding -- we don't overrde unique_id in entities that don't have a way to return unique ids with a dummy method hardcoded to return None.

Reply:
We don't have any base entity properties where we're supposed to both override and delegate/fallback to super. We should be consistent with existing code.

Reply:
Not sure what exactly you mean by base entities, but sure there are some properties that fallback to super. For example min/max_temp in climate.generic_thermostat, battery_level in vacuum.ecovacs, state in climate.sensibo to name a few (and there are more). To me it looks like these are all doing the right thing, and changing them to do otherwise would be making things worse.

Reply:
I mean the base `Entity` class:
https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/helpers/entity.py#L59